File: programmers/level2/카펫.py
def solution(brown, yellow):
    y_wh = (brown-4)//2 #노란색의 둘레 길이//2
    for y_w in range(y_wh, 0, -1): #가로*세로 = 개수
        y_h = y_wh - y_w
        if y_h*y_w == yellow:
            return [y_w+2, y_h+2] 

File: programmers/level2/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int halfRectLength = (brown-4)/2;
        for(int x=halfRectLength-1; x>=1; x--) {
            int y = halfRectLength - x;
            if(x * y == yellow) {
                answer[0] = x+2;
                answer[1] = y+2;
                break;
            }
        }
        return answer;
    }
}